Validation/src/Validation/__main__.py

```diff
@@ -0,0 +1,97 @@
+"""CLI for comparison plots within LDMX Validation"""
+
+# standard
+import argparse
+import os
+import re
+import logging
+
+# external
+import matplotlib
+# this line allows us to run without an X server connected
+#    basically telling MPL that it will not open a window
+matplotlib.use('Agg')
+
+import matplotlib.pyplot as plt
+import mplhep
+plt.style.use(mplhep.style.ROOT)
+
+# us
+from ._differ import Differ
+from ._file import File
+from ._plotter import plotter
+
+# guard incase someone imports this somehow
+if __name__ == '__main__' :
+    parser = argparse.ArgumentParser("python3 -m Validation",
+        description="""
+        Make comparison plots between different files within a directory.
+
+        The labels of different plots within the directory is controlled by
+        the parameter you choose. The parameters of a file are extracted from
+        the file name by splitting the filename into key-val pairs separated 
+        by underscores (i.e. key1_val1_key2_val2_..._keyN_valN.root). If no
+        parameter is provided, then the first key/val is used.
+        """
+        )
+
+    parser.add_argument('data',help='directory of event and histogram files')
+    parser.add_argument('--log',help='logging level',choices=['info','debug','warn','error'], default='warn')
+    parser.add_argument('--label',help='label for grouping of data, defaults to data directory name')
+    parser.add_argument('--out-dir',help='directory to which to print plots. defaults to input data directory')
```

Comment:
I think this is a typo? The out directory is later accessed as out_dir

Reply:
argparse converts `-` to `_` so that the command-line flag is easier to type and the resulting variable can be accessed in Python.

Reply:
For future reference:

[argparse docs](https://docs.python.org/dev/library/argparse.html#dest)

> Any internal - characters will be converted to _ characters to make sure the string is a valid attribute name.

Reply:
I see, then I was doing something else wrong when I ran into this :)

Reply:
Were you seeing the plots not go into the selected output directory? There could still be a bug lurking